lib/AST/Decl.cpp

```diff
@@ -3900,12 +3898,7 @@ GetDestructorRequest::evaluate(Evaluator &evaluator, ClassDecl *CD) const {
 
 
 bool ClassDecl::hasMissingDesignatedInitializers() const {
-  auto *mutableThis = const_cast<ClassDecl *>(this);
-  auto flags = OptionSet<LookupDirectFlags>();
-  flags |= LookupDirectFlags::IgnoreNewExtensions;
-  (void)mutableThis->lookupDirect(DeclBaseName::createConstructor(),
-                                  flags);
-  return Bits.ClassDecl.HasMissingDesignatedInitializers;
+  return getAttrs().hasAttribute<HasMissingDesignatedInitsAttr>();
```

Comment:
This still needs to do the `lookupDirect`, because forcing the inits to be loaded is what determines if any are missing in a compiled module (and soon the importer too).